numpyro/infer/mcmc.py

```diff
@@ -732,20 +740,19 @@ def _single_chain_mcmc(self, rng_key, init_params, args, kwargs,
             if self._warmup_state is None:
                 raise ValueError('No `init_state` found; warmup results cannot be reused.')
             num_warmup = 0
-            init_state = self._warmup_state
+            init_state = self._warmup_state._replace(rng_key=rng_key)
```

Comment:
This was a bug! We were basically returning the same result as the previous run.

Reply:
I think for reproducible purpose, it would be nice to allow `rng_key` have a batch_size (in that case, we won't split). Previously, with functional interface, we can verify the consistency with
```
mcmc_1chain.run(random.PRNGKey(0), ...)
mcmc_4chains.run(vmap(random.PRNGKey)(range(4)), ...)
```
Similarly, here, we can verify the consistency of `reuse_warmup_state` by setting
```
mcmc.run(mcmc._warmup_state.rng_key, ...)
```